### PR TITLE
Refactory: Remove unnecessary use of ramda and make code more testable

### PR DESCRIPTION
**Pull request by klaygomes**
Today you are using an in-memory db, but in future if you want/need change this aproach it will be necessary to rewrite all tests (or use a local db), anyway I don't think it is a good ideia mix business with data access. 

I made this PR just for us to discuss about it.

### CODE REVIEW

File: src/game/core/explore.js [hunk 1]
```diff
@@ -11,24 +10,27 @@ import {
...
 import { run } from './combat'
 
-export function randomMonster (mapId) {
-  const mapObj = models.maps.find(mapId)
-  const monsterPool = mapObj.monsters.reduce((acc, monster) => [
+export function randomMonster (monsters, pickMonsterFn=monsters.find) {
+  const monsterPool = monsters.reduce((acc, monster) => [
     ...acc,
-    ...Array.from({ length: monster.influence }).map(always(monster.id)),
+    ...Array.from({ length: monster.influence }, ()=> monster.id ),
```
Comment (MarcoWorms) [inline src/game/core/explore.js]: so good!

File: src/game/core/explore.js [hunk 2]
```diff
@@ -11,24 +10,27 @@ import {
...
+  maps,
+  monsters,
+} from '../models'
+
 import { run } from './combat'
 
-export function randomMonster (mapId) {
-  const mapObj = models.maps.find(mapId)
-  const monsterPool = mapObj.monsters.reduce((acc, monster) => [
+export function randomMonster (monsters, pickMonsterFn=monsters.find) {
```
Comment (MarcoWorms) [inline src/game/core/explore.js]: awesome use of default parameters :D

**Comment by derekstavis** [comment]
Hey! Thanks for the PR and the discussion opportunity :smile:

About this PR: Good job! I loved the `Array.from` solution, default parameter trick, very :+1:

There are minor things: I would avoid comments. I think that if we need to comment code is because it's not clear, so we should code clearer instead of using comments. How we do this? We use Ramda :ok_hand: and that leds to point two: I would keep the `always` function instead of using arrows. We are willing to push Ramda wherever we can use to be more clear about what code does. `always(value)` says a lot more than `() => value`.

Now about the game architecture:

>I don't think it is a good ideia mix business with data access.

This was by design. The premise is that models are considered part of game runtime. Keeping stuff in memory and fitting game business logics directly in model data fits very well in code. It avoids glue and holds back code fragmentation, something very important for easing contribution curve for people that are new to coding. This is the idea behind having a good, POJO model interface.

**Some future thoughts about refactors I wanted to do:**

My initial idea was to promisify all of `models/` directory, and make them share the very same `dao/` interface (as far as possible). Also, I see a lot of custom errors working fluidly in the whole project, because Bluebird custom errors are fucking nice 🌟 haha

There are still a lot of refactors to do. I will be working next days in splitting view and handler logic because it's making me sick, and also moving code away from handlers to core. We rushed to deliver and now we need to pay technical debt.

Any help is accepted and we are open to discuss and improve each others architecture. Now that we've published the initial version we've set our bars high, and we will try a lot to keep PR quality, good code cohesion and standards. We will try hard to have a good test coverage too.

So, to finish. Thanks for observing and following commit standard. Addressed the points mentioned above (comments and ramda usage), I'm very :+1: to merging this! :tada: 

Thanks again for the PR! I'm really excited about your contribution :)

File: src/game/core/explore.js (hunk 1 above)
Comment (derekstavis) [inline src/game/core/explore.js]: `always(monster.id)` reads clearer than `() => monster.id` :stuck_out_tongue:

Comment (MarcoWorms) [inline src/game/core/explore.js]: i agree, `...Array.from({ length: monster.influence }, always(monster.id) )` looks better

Comment (klaygomes) [inline src/game/core/explore.js]: Well, I don't think it is less readable, anyone knows what a fat arrow is and how barely it works, but nor radma. I had to look at documentation to understand what always returns back for example. That is why I put it into our discussion :)

Comment (derekstavis) [inline src/game/core/explore.js]: Understanding Ramda is a prerequisite for messing with project's core, so Ramda knowledge shouldn't be an argument. About readability, what I mean is that in plain english `always(monster)` is more expressive than `() => monster`, that's why we prefer using Ramda where possible :smile:

**Comment by klaygomes** [comment]
> We are willing to push Ramda wherever we can use to be more clear about what code does. always(value) reads much better than () => value.

Good point! But I think it is a win loss. This design aproach will require developers know ramda besides vanilha js. But I'm with you that it should be used always when it makes code more readable.

> This was by design. The premise is that models are considered part of game runtime. Keeping stuff in memory and fitting game business logics directly in model data fits very well in code. It avoids glue and holds back code fragmentation, something very important for easing contribution curve for people that are new to coding. This is the idea behind having a good, POJO model interface.

I didn't get it. Maybe it is still too early for me to see what you are planning for. 🤞 

> Any help is accepted and we are open to discuss and improve each others architecture. Now that we've published the initial version we've set our bars high, and we will try a lot to keep PR quality, good code cohesion and standards. We will try hard to have a good test coverage too.

I see, I will keep watching this project while you get paid your technical debt, I hope in future I can contribute in a better way.  😄

File: src/game/core/explore.js (hunk 1 above)
Comment (haskellcamargo) [inline src/game/core/explore.js]: Well, my unique opinion is about coding-style that the linter will catch. In this case, `()=>` to `() =>`.

File: src/game/core/explore.js (hunk 2 above)
Comment (haskellcamargo) [inline src/game/core/explore.js]: Taking account we are currently implementing a linting system, `pickMonsterFn=monsters.find` will break futurely. `pickMonsterFn=monsters.find` `->` `pickMonsterFn = monsters.find`.

**Comment by derekstavis** [comment]
Closing by inactivity.
